modules/tinymce/src/themes/silver/test/ts/browser/throbber/ThrobberFocusTest.ts

```diff
@@ -0,0 +1,175 @@
+    TinyUiActions.pWaitForDialog(editor);
+    await pAssertDialogFocus();
+
+    // Throbber should not steal focus from the dialog
+    await pEnableThrobber(editor, pAssertDialogFocus);
+
+    // Throbber should get focus when dialog is closed instead of editor
+    TinyUiActions.cancelDialog(editor);
+    await Waiter.pTryUntil('Dialog should close', () => UiFinder.notExists(SugarBody.body(), 'div[role="dialog"]'));
+    await pAssertThrobberFocus();
+
+    // Focus returns to the editor when throbber is closed (this should happen since a dialog will normally focus the editor on close)
+    await pDisableThrobber(editor);
+    assert.isTrue(editor.hasFocus());
+  });
+
+  // In theory, this combination shouldn't be possible as the user cannot click any buttons to open dialogs when the throbber is enabled
+  // However, it is still important that all of the focus transitions make sense
+  it('TINY-7373: should not set focus on the editor if the throbber did not have focus when disabled', async () => {
+    const editor = hook.editor();
+    const pAssertDialogFocus = () => FocusTools.pTryOnSelector('Dialog input has focus (1)', SugarDocument.getDocument(), 'input.tox-textfield');
+
+    await pEnableThrobber(editor);
+    openDialog(editor);
+    TinyUiActions.pWaitForDialog(editor);
+    await pAssertDialogFocus();
+
+    // Disable the throbber when the dialog is open
+    await pDisableThrobber(editor, pAssertDialogFocus);
+
+    // Focus returns to the editor when the dialog is closed the throbber is not enabled
+    TinyUiActions.cancelDialog(editor);
+    await Waiter.pTryUntil('Dialog should close', () => UiFinder.notExists(SugarBody.body(), 'div[role="dialog"]'));
+    assert.isTrue(editor.hasFocus());
+  });
+
+  it('TINY-7373: should have correct focus transitions when opening and closing notification', async () => {
+    const editor = hook.editor();
+    const pAssertNotificationFocus = () =>
+      FocusTools.pTryOnSelector('Notification close button has focus', SugarDocument.getDocument(), 'button.tox-notification__dismiss');
+    const notification = editor.notificationManager.open({
+      text: 'Test',
+      closeButton: true
+    });
+    const popup = await TinyUiActions.pWaitForPopup(editor, 'div.tox-notification') as SugarElement<HTMLElement>;
+    Focus.focus(popup);
+    await pAssertNotificationFocus();
+
+    // Throbber should not steal focus from the notification
+    await pEnableThrobber(editor, pAssertNotificationFocus);
+
+    // Throbber should get focus when notification is closed instead of editor
+    notification.close();
+    await Waiter.pTryUntil('Notification should close', () => UiFinder.notExists(SugarBody.body(), 'div.tox-notification'));
+    await pAssertThrobberFocus();
+
+    // Focus returns to the editor when throbber is closed (this should happen since a dialog will normally focus the editor on close)
+    await pDisableThrobber(editor);
+    assert.isTrue(editor.hasFocus());
+  });
```

Comment:
- [ ] Can you add some tests to make sure that users clicking on / tabbing into the editor doesn't work while the throbber is up

Reply:
Thanks for the suggestion. I have added the tabbing test to make sure that works correctly. I think a clicking test would be better served as a manual test as it should not be possible for the user to click the editor since the throbber div covers the whole editor. In order to test clicking would require testing a scenario that user couldn't actually do themselves.